docs/decisions/2027-07-26_agent_command_commands.md

```diff
@@ -0,0 +1,18 @@
+# 2023-07-26 Agent Task Execution and Command Graceful Exit
+
+* status: accepted
+* date: 2023-07-26
+* authors: Kim Tao
+
+## Context and Problem Statement
+
+The agent was implemented run pre and main block commands in a separate goroutine and then waited on either it to
```

Comment:
*to run

Reply:
I reworked this sentence a little to have less awkward phrasing.